src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java

```diff
@@ -52,6 +57,24 @@ public void collectMetrics(long startTime) {
 
         while (statsIterator.hasNext()) {
             Stats stats = statsIterator.next();
+            long rejectionDelta = 0;
+            String name = stats.getName();
+            if (statsRecordMap.containsKey(name)) {
+                ThreadPoolStatsRecord lastRecord = statsRecordMap.get(name);
+                // if the timestamp in previous record is greater than 15s (3 * intervals),
+                // then the scheduler might hang or freeze due to long GC etc. We simply drop
+                // previous record here and set rejectionDelta to 0.
+                if (startTime - lastRecord.getTimestamp() <= SAMPLING_TIME_INTERVAL * 3) {
```

Comment:
Consider moving the  interval count [i.e. 3 here] to a variable in config.

Reply:
The reason why I hardcoded this to 3 is because this collector is scheduled every 5 second  is very unlikely to go beyond 3*5 second unless something unexpected happens (system hangs etc.). And in that case we simply drop the data if this happens. So I don't think we need to expose this in config file because this is not something we want to tune in runtime

Reply:
Makes sense.